packages/cardano-services/test/Asset/CardanoTokenRegistry.test.ts

```diff
@@ -0,0 +1,46 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { CardanoTokenRegistry } from '../../src/Asset';
+import { dummyLogger } from 'ts-log';
+
+describe('CardanoTokenRegistry', () => {
+  const tokenRegistry = new CardanoTokenRegistry({ logger: dummyLogger });
+  const nonvalidAssetId = Cardano.AssetId('0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef');
+  const validAssetId = Cardano.AssetId('f43a62fdc3965df486de8a0d32fe800963589c41b38946602a0dc53541474958');
+
+  it('returns null for nonexisting AssetId', async () => {
+    expect(await tokenRegistry.getTokenMetadata([nonvalidAssetId])).toEqual([null]);
+  });
+
+  it('returns metadata when subject exists', async () => {
+    const [metadata] = await tokenRegistry.getTokenMetadata([validAssetId]);
+
+    expect(metadata).not.toBeNull();
+    if (!metadata) return;
+
+    // Skip icon from test as it is a long long string
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { icon, ...rest } = metadata;
+    const result: Asset.TokenMetadata = {
+      decimals: 8,
+      // eslint-disable-next-line max-len
+      desc: "SingularityNET lets anyone - create, share, and monetize AI services at scale. SingularityNET is the world's first decentralized AI network",
+      name: 'SingularityNet AGIX Token',
+      ticker: 'AGIX',
+      url: 'https://singularitynet.io/'
+    };
+
+    expect(rest).toEqual(result);
+  });
+
+  it('returns cached metadata for already requested subject', async () => {
+    // We need to get metadata at the beginning of the test, to make work
+    // yarn test -t "returns cached metadata for already requested subject"
```

Comment:
It would be better to spy on methods that access the cache, which will be easier after implementing `InMemoryCache`